### ISSUE TITLE
Better node boundary box identification

### ISSUE THREAD

**Issue by epuscasu** [godotengine/godot#30012]
There might be some confusion when working with certain nodes that involve placing control points (such as Path2D) and the node's boundary box:

![image](https://user-images.githubusercontent.com/11788042/59980655-94f92380-9601-11e9-9a8d-e5907f496618.png)

This is what a newly created Path2D node looks like - is the red dot a curve control point? Not knowing beforehand, I'd say yes, but creating a new Point makes the original one disappear. Or, if grid snap is enabled, a new one appears alongside the old one (which isn't even a control point)

![image](https://user-images.githubusercontent.com/11788042/59980700-29638600-9602-11e9-8172-097a9d51d24a.png)

Continuing to add more points leads to even more of them showing up, often overlapping and are quite confusing.

![image](https://user-images.githubusercontent.com/11788042/59980707-4435fa80-9602-11e9-8cbe-1223fdba5bc6.png)

![image](https://user-images.githubusercontent.com/11788042/59980709-4ac47200-9602-11e9-9445-8b2ed4c4eec3.png)
![image](https://user-images.githubusercontent.com/11788042/59980714-557f0700-9602-11e9-9aa5-6577e92cba67.png)

I think a better way of handling this would be differently colored / shaped boundary rectangle points or the option of disabling it.

**Comment by cbscribe** [comment]
Agreed, this is very confusing. The control points should be hidden as they are in CollisionShape2D, etc.
